### DISCUSSION TITLE
Can V be used for creating shared libraries?

### DISCUSSION THREAD

**Discussion by alpominth** [vlang/v#18282] [Questions and Answers] [answered]
I searched a lot in Google about creating shared libraries in V but I didn't see anything about that.

Can V be used for creating shared libraries like C does?

**Comment by JalonSolov** [comment, marked as answer]
Add the `-shared` option to the compile line.  It will create a `.so` on Unix, a `.dll` on Windows, etc.

**Reply by alpominth** [reply]
Very nice.

I have been wondering if V could possibly be used to create .ko kernel files, as it supports GCC backend.

**Reply by JalonSolov** [reply]
I don't know if it creates a `.ko` file, but check out https://github.com/spytheman/simple_kernel_module_in_v
